aten/src/ATen/native/SpectralOps.cpp

```diff
@@ -409,6 +409,66 @@ Tensor fft_irfftn(const Tensor& self, c10::optional<IntArrayRef> s,
   return native::fft_irfft(x, last_shape, last_dim, norm);
 }
 
+Tensor fft_fftfreq(int64_t n, double d, const TensorOptions& options) {
+  ScalarType dtype = typeMetaToScalarType(options.dtype());
+  TORCH_CHECK(at::isFloatingType(dtype) || at::isComplexType(dtype),
+              "fftfreq requires a floating point or complex dtype");
+  Tensor result = native::arange(n, options);
+  auto right_slice = result.slice(0, (n + 1) / 2, 0);
+  at::arange_out(right_slice, -(n/2), 0, 1);
+  result.mul_(1.0 / (n * d));
```

Comment:
Maybe a brief note here explaining why the multiplication by the reciprocal is occurring instead of a call to `true_divide_`.